### PR TITLE
Add a target definition file as a new module

### LINKED ISSUE #592: Add target definition module for later use by tycho
To speed up the build process with tycho and to be able to use the same target definition in eclipse that tycho uses, a target definition module should be added according to the following docs: https://wiki.eclipse.org/Tycho/Packaging_Types#eclipse-target-definition

### PR DESCRIPTION
**Pull request by Flugtiger**
fixes #592

### CODE REVIEW

File: build/org.eclipse.birt.target/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/target/
```
Comment (ruspl-afed) [inline build/org.eclipse.birt.target/.gitignore:1]: I think this is good to have for `.gitignore` in the root

File: build/org.eclipse.birt.target/pom.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project
```
Comment (ruspl-afed) [inline build/org.eclipse.birt.target/pom.xml:2]: Please add EPL 2.0 header

File: build/org.eclipse.birt.target/org.eclipse.birt.target.target [hunk 3]
```diff
@@ -0,0 +1,112 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<?pde version="3.8"?>
+<target includeMode="feature" name="BIRT Photon target platform">
```
Comment (ruspl-afed) [inline build/org.eclipse.birt.target/org.eclipse.birt.target.target:3]: Please add EPL 2.0 header

File: build/org.eclipse.birt.target/org.eclipse.birt.target.target [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+	<locations>
+		<location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
+			<repository location="http://download.eclipse.org/eclipse/updates/4.11/"/>
```
Comment (ruspl-afed) [inline build/org.eclipse.birt.target/org.eclipse.birt.target.target:6]: Please prefer https

File: build/org.eclipse.birt.target/org.eclipse.birt.target.target [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+			<unit id="org.eclipse.platform.sdk" version="0.0.0"/>
+			<unit id="org.eclipse.equinox.executable.feature.group" version="0.0.0"/>
+			<unit id="org.eclipse.cvs.feature.group" version="0.0.0"/>
+			<unit id="org.eclipse.rcp.configuration.feature.group" version="0.0.0"/>
+			<unit id="a.jre.javase" version="0.0.0"/>
+		</location>
+		<location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
+			<repository location="http://download.eclipse.org/modeling/emf/emf/builds/milestone/latest"/>
```
Comment (ruspl-afed) [inline build/org.eclipse.birt.target/org.eclipse.birt.target.target:14]: Please prefer https

Comment (ruspl-afed) [inline build/org.eclipse.birt.target/org.eclipse.birt.target.target:14]: IMO it is better to reference the EMF that was a part of 4.11 to be consistent, and not the latest one.

File: build/org.eclipse.birt.target/org.eclipse.birt.target.target [hunk 6]
```diff
@@ -0,0 +1,112 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<?pde version="3.8"?>
+<target includeMode="feature" name="BIRT Photon target platform">
+	<locations>
+		<location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
+			<repository location="http://download.eclipse.org/eclipse/updates/4.11/"/>
+			<unit id="org.eclipse.platform.sdk" version="0.0.0"/>
+			<unit id="org.eclipse.equinox.executable.feature.group" version="0.0.0"/>
+			<unit id="org.eclipse.cvs.feature.group" version="0.0.0"/>
```
Comment (ruspl-afed) [inline build/org.eclipse.birt.target/org.eclipse.birt.target.target:9]: do we support CVS explicitly?

File: build/org.eclipse.birt.target/org.eclipse.birt.target.target [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+		<location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
+			<repository location="https://download.eclipse.org/webtools/downloads/drops/R3.13.0/R-3.13.0-20190308131645/repository/"/>
+			<unit id="org.eclipse.jst.enterprise_ui.feature.feature.group" version="0.0.0"/>
+			<unit id="org.eclipse.wst.jsdt.feature.feature.group" version="0.0.0"/>
+			<unit id="org.eclipse.jst.web_ui.feature.feature.group" version="0.0.0"/>
+			<unit id="org.eclipse.wst.web_ui.feature.feature.group" version="0.0.0"/>
+			<unit id="org.eclipse.jst.common.fproj.enablement.jdt.feature.group" version="0.0.0"/>
+		</location>
+		<location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
+			<repository location="http://download.eclipse.org/jetty/updates/jetty-bundles-9.x/9.4.15.v20190215"/>
```
Comment (ruspl-afed) [inline build/org.eclipse.birt.target/org.eclipse.birt.target.target:66]: Please prefer https

File: build/org.eclipse.birt.target/org.eclipse.birt.target.target [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+			<unit id="org.eclipse.jetty.server" version="0.0.0"/>
+			<unit id="org.eclipse.jetty.util" version="0.0.0"/>
+			<unit id="org.eclipse.jetty.webapp" version="0.0.0"/>
+			<unit id="org.eclipse.jetty.plus" version="0.0.0"/>
+			<unit id="org.eclipse.jetty.servlet" version="0.0.0"/>
+			<unit id="org.eclipse.jetty.http" version="0.0.0"/>
+		</location>
+		<location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
+			<unit id="org.apache.jasper" version="0.0.0"/>
+			<repository location="http://download.eclipse.org/gemini/updates/web/3.0.4"/>
```
Comment (ruspl-afed) [inline build/org.eclipse.birt.target/org.eclipse.birt.target.target:77]: Please prefer https

Comment (ruspl-afed) [inline build/org.eclipse.birt.target/org.eclipse.birt.target.target:77]: Let's always declare `repository` before any `unit`, like it is done above and below

File: build/org.eclipse.birt.target/org.eclipse.birt.target.target [hunk 9]
```diff
@@ -0,0 +1,112 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<?pde version="3.8"?>
+<target includeMode="feature" name="BIRT Photon target platform">
+	<locations>
```
Comment (ruspl-afed) [inline build/org.eclipse.birt.target/org.eclipse.birt.target.target:4]: I suggest to sort `location` elements according to its `repository.location` attribute value
   and to sort `unit` elements according to its `id` value
That could give clear (and automatable) normalization rule for .target content.

**Review by ruspl-afed — CHANGES_REQUESTED** [review]
we need to include this module to a build to check the added target for consistency

File: build/org.eclipse.birt.target/org.eclipse.birt.target.target (hunk 5 above)
Comment (Flugtiger) [reply build/org.eclipse.birt.target/org.eclipse.birt.target.target:14]: Which one was it? When I created this target platform file I only copied the repositories from the parent-pom.

File: build/org.eclipse.birt.target/org.eclipse.birt.target.target (hunk 6 above)
Comment (Flugtiger) [reply build/org.eclipse.birt.target/org.eclipse.birt.target.target:9]: I guess so

**Comment by ruspl-afed** [comment]
@Flugtiger it looks like we should add more excludes for orbit bundles to the configuration of `tycho-source-feature-plugin`

File: build/org.eclipse.birt.target/org.eclipse.birt.target.target (hunk 6 above)
Comment (wimjongman) [reply build/org.eclipse.birt.target/org.eclipse.birt.target.target]: That sounds strange. CVS? :)

**Comment by Flugtiger** [comment]
> @Flugtiger it looks like we should add more excludes for orbit bundles to the configuration of `tycho-source-feature-plugin`

Do you know why it can't resolve the sources? When building without the target definition, it can find them.

**Comment by Flugtiger** [comment]
Nevermind, I found it: we need `<targetDefinitionIncludeSource>honor</targetDefinitionIncludeSource>` in the tycho-platform-plugin config.

**Comment by Flugtiger** [comment]
The report designer depends on org.eclipse.cvs:
image

**Comment by ruspl-afed** [comment]
> The report designer product (build/birt-packages/birt-report-all-in-one/BIRT.product) depends on org.eclipse.cvs:
> could this be a mistake?

Well it could be, but who knows what is going on in the dark deep corporate environment :)
I suggest to manage this archaeological issue separately.

File: pom.xml [hunk 10]
```diff
@@ -371,6 +373,14 @@
 				<version>${tycho.version}</version>
 				<configuration>
 					<pomDependencies>consider</pomDependencies>
+					<targetDefinitionIncludeSource>honor</targetDefinitionIncludeSource>
```
Comment (ruspl-afed) [inline pom.xml:376]: Bravo!

**Comment by ruspl-afed** [comment]
Amazing work @Flugtiger , I hope to see it merged soon.

**Pull request merged**